docroot/modules/humsci/hs_field_helpers/src/Plugin/views/query/Sql.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Drupal\hs_field_helpers\Plugin\views\query;
+
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\views\Plugin\views\query\Sql as OriginalSql;
+use Drupal\views\ViewExecutable;
+
+/**
+ * Class Sql that adds extra functionality to views Sql.
+ *
+ * @package Drupal\hs_field_helpers\Plugin\views\query
+ */
+class Sql extends OriginalSql {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAggregationInfo() {
+    $info = parent::getAggregationInfo();
+    $info['greatest'] = [
+      'title' => $this->t('Greatest'),
+      'method' => 'aggregationMethodSimple',
+      'handler' => [
+        'argument' => 'groupby_numeric',
+        'field' => 'numeric',
+        'filter' => 'groupby_numeric',
+        'sort' => 'groupby_numeric',
+      ],
+    ];
+    $info['least'] = [
+      'title' => $this->t('Least'),
+      'method' => 'aggregationMethodSimple',
+      'handler' => [
+        'argument' => 'groupby_numeric',
+        'field' => 'numeric',
+        'filter' => 'groupby_numeric',
+        'sort' => 'groupby_numeric',
+      ],
+    ];
+    return $info;
+  }
+
+  /**
+   * Alter the view query in consolidate multiple date field sorting.
+   *
+   * @param \Drupal\views\ViewExecutable $view
+   *   View object for the query.
+   */
+  public function alterQuery(ViewExecutable $view) {
+    $date_fields = $this->getDateFields();
+
+    // Not enough date fields to care about altering.
+    if (!(count($date_fields) > 1)) {
+      return;
+    }
+
+    $expression = [];
+    $orderby_delta = -1;
+
+    // Build the expression.
+    foreach ($date_fields as $field_alias => $direction) {
+      $field = $this->fields[$field_alias];
+
+      // Set null values to 0.
+      $expression[] = 'COALESCE(' . $field['table'] . '.' . $field['field'] . ', 0)';
```

Comment:
The `COALESCE` function seems to be valid for PGSQL, MySQL, and MSSQL. Not sure if there are slight behavior differences.

Reply:
just noting?